sharding/mainchain/interfaces.go

```diff
@@ -63,5 +63,6 @@ type RecordFetcher interface {
 		ChunkRoot [32]byte
```

Comment:
Hey rather than defining it as inline structs can you define it in `contracts/types.go` and use it from there. It is similar to what I have done in the file. Or if it is out of scope for this pr, we can put this as a separate issue

Reply:
I don't think it will work because collationRecords is originated from `sharding_manager.go` and we shouldn't modify that file. It will break `SMCCaller`

In `sharding_manager.go`
```
func (_SMC *SMCCaller) CollationRecords(opts *bind.CallOpts, arg0 *big.Int, arg1 *big.Int) (struct {
	ChunkRoot [32]byte
	Proposer  common.Address
	IsElected bool
	Signature []byte
}, error)
```

Reply:
No I didn't mean to modify collationRecords in `sharding_manager.go`. I meant defining the return value  of it as a type in `contracts/types.go` as 
```go
var record struct {
	ChunkRoot [32]byte
	Proposer  common.Address
	IsElected bool
	Signature []byte
}
```
Then instead of defining the struct inline all the time whenever you need to use it you can just use `*contracts.record`.  You can take a look at `contracts/types.go` for an example

Reply:
The following are different signatures and it'll break SMCCaller if we use types.CollationRecord

```
	CollationRecords(opts *bind.CallOpts, arg0 *big.Int, arg1 *big.Int) (struct {
		ChunkRoot [32]byte
		Proposer  common.Address
		IsElected bool
		Signature []byte
	}, error)
```

```
	CollationRecords(opts *bind.CallOpts, arg0 *big.Int, arg1 *big.Int) (types.CollationRecord, error)
```